quinn-proto/src/connection/mod.rs

```diff
@@ -3920,6 +3929,10 @@ pub enum Event {
     DatagramReceived,
     /// One or more application datagrams have been sent after blocking
     DatagramsUnblocked,
+    /// One or more TLS session resumption tickets have been received
+    ///
+    /// This event is only emitted on the client, and is emitted at most once per connection.
+    ResumptionTicketsReceived,
```

Comment:
This seems sort of specific to the TLS implementation -- our crypto traits are intended to be more general than that. Maybe this should be renamed to `ResumptionEnabled`?

Also, note that `Event` is not `non_exhaustive`, so this constitutes a semver-incompatible change. In general, this feels fairly invasive for what seems like a somewhat niche feature. Maybe there is some other way to structure this?

Reply:
WDTY - Should I keep the new event, or add a getter that is checked on each poll? Or did you have something else in mind for how to forward the event from `quinn-proto` to `quinn`?

Reply:
For Quinn to provide an async API, polling would need to happen on the critical path to trigger a `Notify` or similar, which strikes me as almost as invasive. An event is more plumbing, but doesn't add a branch that gets checked on every poll forever. If we're going to have this, I lean slightly towards an event, but I don't feel very strongly.